python/lsst/ts/watcher/mock_opsgenie.py

```diff
@@ -0,0 +1,217 @@
+        self.reject_next_request = False
+
+    def make_app(self) -> web.Application:
+        """Make an instance of the web application."""
+        app = web.Application()
+        app.add_routes(
+            [
+                web.post("/:{identifier}/close", self.close_alert),
+                web.post("/", self.create_alert),
+                web.delete("/:{identifier}", self.delete_alert),
+            ]
+        )
+        return app
+
+    async def start(self) -> None:
+        """Start the service.
+
+        Raises
+        ------
+        RuntimeError
+            If port = 0 and serving on more than one socket
+            (in which case the served port cannot be determined).
+        """
+        if self.runner is not None:
+            raise RuntimeError("Already started")
+        app = self.make_app()
+        self.runner = web.AppRunner(app)
+        await self.runner.setup()
+        self.site = web.TCPSite(self.runner, "127.0.0.1", port=self.port)
+        await self.site.start()
+        server = self.site._server
+        if self.port == 0:
+            if len(server.sockets) != 1:
+                raise RuntimeError(
+                    "Serving on more than one socket; cannot determine the port"
+                )
+            self.port = server.sockets[0].getsockname()[1]  # type: ignore
+        self.url = f"http://127.0.0.1:{self.port}"
+
+    async def cleanup(self) -> None:
+        """Stop the service."""
+        if self.runner is None:
+            raise RuntimeError("No runner; did you call start?")
+        await self.runner.cleanup()
+
+    def assert_authorized(self, request) -> None:
+        """Raise an error if or reject_next_request true or not authorized.
+
+        Raise HTTPInternalServerError if reject_next_request True
+        after resetting that flag.
+        Raise HTTPForbidden if request is not authorized.
+        """
+        if self.reject_next_request:
+            self.reject_next_request = False
+            raise web.HTTPInternalServerError(
+                text="Failed by request: reject_next_request true"
+            )
+
+        auth_header = request.headers.get("Authorization", None)
+        if auth_header is None:
+            raise web.HTTPForbidden(text="No Authorization header")
+        try:
+            name, value = auth_header.split(None, 1)
+        except ValueError:
+            raise web.HTTPForbidden(
+                text="Authorization header must be of the form 'GenieKey <key>'"
+            )
+        if name != "GenieKey":
+            raise web.HTTPForbidden(
+                text="Authorization header must start with GenieKey"
+            )
+        if value != self.escalation_key:
+            raise web.HTTPForbidden(text="Authorization key does not match")
+        return None
+
+    async def close_alert(self, request) -> web.json_response:
+        """Request handler to set the status of an alert to "closed"."""
```

Comment:
Can you please complete the doc string for this function and describe the `request` parameters as well as the return value. Also for `create_alert` and `delete_alert` please.

Reply:
Yes. I'll add the missing type annotation, as well. I don't know how to make the  description very informative, but it's still better to have it.